### PR TITLE
#1526 november dependency upgrades

### LINKED ISSUE #1526: Dependency upgrades November 2019
## Is your feature request related to a problem? Please describe.

Monthly dependency upgrade

## Describe the solution you'd like

- Mainly looking to upgrade `react-navigation`. It was left out of last months as there were other major ones and didn't want to do too many at once. All of the navigators have been ripped out of the main package and are their individual packages now - i.e. stack/switch/tabs. Not a biggy as they've made them backwards compatible I believe, but a couple of code changes, I believe
- Haven't seen any other major releases

## Describe alternatives you've considered

Getting behind in upgrades

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1526 

## Change summary

- upgrade `react-native-navigation` [this new version is a breaking change, where every navigator has been extracted out of the package]
- Added `react-native-stack` [this is the extracted package from above]
- Upgrade/bump minor react version [nothing special]
- Upgrade react-native-device-info [Something with another new androidID. Not 100% sure]
- react-native-fs [bug fixes]
- react-native-gesture-handler [bug fixes]
- async-storage [bug fixes]
- react-native [bug fixes]
- redux-helpers [bug fixes]
- react-redux [bug fixes]

## Testing

Never know what to put here..

- [ ] Can run `react-native run-android`
- [ ] Can run `yarn build`

### Related areas to think about

Still trying to fix apk's :(

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,30 +53,34 @@
...
   "dependencies": {
-    "@react-native-community/async-storage": "^1.5.0",
+    "@react-native-community/async-storage": "^1.6.2",
     "@react-navigation/core": "^3.5.1",
     "bugsnag-react-native": "2.23.2",
+    "lodash": "4.17.15",
     "prop-types": "^15.7.2",
-    "react": "^16.10.2",
-    "react-native": "0.61.2",
+    "react": "16.9.0",
```
Comment (Chris-Petty) [inline package.json]: downgrade?

File: package.json [hunk 2]
```diff
@@ -53,30 +53,34 @@
     ]
   },
   "dependencies": {
-    "@react-native-community/async-storage": "^1.5.0",
+    "@react-native-community/async-storage": "^1.6.2",
     "@react-navigation/core": "^3.5.1",
     "bugsnag-react-native": "2.23.2",
+    "lodash": "4.17.15",
```
Comment (Chris-Petty) [inline package.json]: Not sure why introduced, is it a peer dependancy for something?

Comment (joshxg) [reply package.json]: Mustv'e slipped through. I didn't add it explictly so mustv'e been a peer dependency. I'll try and look or just revert?

File: package.json (hunk 1 above)
Comment (joshxg) [reply package.json]: Just the version RN wants - mainly did for peer dependency warning. Just bug fixes that I don't believe eeffect us

File: package.json (hunk 2 above)
Comment (Chris-Petty) [reply package.json]: I don't think it can get in there by itself? Just quickly remove and check if it whines?

Comment (joshxg) [reply package.json]: 41aa639

**Pull request merged**